hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/logic/StandardProcessLogic.java

```diff
@@ -82,48 +83,51 @@ public StandardProcessLogic(
-                return;
-            }
+            incorporate(accessor, platformTxn.getConsensusTimestamp(), submittingMember);
+        } catch (InvalidProtocolBufferException e) {
+            log.warn("Consensus platform txn was not gRPC!", e);
+        } catch (Exception internal) {
+            log.error("Unhandled internal process failure", internal);
+        }
+    }
 
-            consensusTimeTracker.reset(consensusTime);
+    void incorporate(
+            final SwirldsTxnAccessor accessor, Instant consensusTime, final long submittingMember) {
+        // Deduct 1000 nanos from the consensusTime allotted by platform, to accommodate the
+        // preceding,
+        // following child records and any long term scheduled transactions triggered by the
+        // current transaction
+        // in the balance file with consensus timestamp X to include all transactions whose
+        // consensus time T <= X.
+        consensusTime = consensusTime.minusNanos(MIN_TRANS_TIMESTAMP_INCR_NANOS);
+
+        accessor.setStateView(workingView);
+        if (!invariantChecks.holdFor(accessor, consensusTime, submittingMember)) {
+            return;
+        }
 
-            sigImpactHistorian.setChangeTime(consensusTime);
-            expiries.purge(consensusTime.getEpochSecond());
-            sigImpactHistorian.purge();
-            recordStreaming.resetBlockNo();
+        consensusTimeTracker.reset(consensusTime);
 
-            doProcess(
-                    submittingMember,
-                    consensusTimeTracker.isFirstUsed()
-                            ? consensusTimeTracker.nextTransactionTime(true)
-                            : consensusTimeTracker.firstTransactionTime(),
-                    accessor);
+        sigImpactHistorian.setChangeTime(consensusTime);
+        expiries.purge(consensusTime.getEpochSecond());
+        sigImpactHistorian.purge();
+        recordStreaming.resetBlockNo();
 
-            if (scheduleProcessing.shouldProcessScheduledTransactions(consensusTime)) {
-                processScheduledTransactions(consensusTime, submittingMember);
-            }
+        doProcess(
+                submittingMember,
+                consensusTimeTracker.isFirstUsed()
+                        ? consensusTimeTracker.nextTransactionTime(true)
+                        : consensusTimeTracker.firstTransactionTime(),
+                accessor);
 
-            autoRenewal.execute(consensusTime);
-            platformTxn.clearSignatures();
-        } catch (InvalidProtocolBufferException e) {
-            log.warn("Consensus platform txn was not gRPC!", e);
-        } catch (Exception internal) {
-            log.error("Unhandled internal process failure", internal);
+        if (scheduleProcessing.shouldProcessScheduledTransactions(consensusTime)) {
+            processScheduledTransactions(consensusTime, submittingMember);
         }
+
+        autoRenewal.execute(consensusTime);
```

Comment:
We don't need to `platformTxn.clearSignatures();` ?

Reply:
Platform holds some recent events (and thus `Transaction` objects) in memory, to use for catching up during reconnect.

But if a `Transaction` **is** used for reconnect, in `develop` branch we [will re-create its signatures in `preHandle()`](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/HederaToPlatformSigOps.java#L77) anyways.

This means that as soon as a `Transaction` reaches consensus, the `develop` code might as well clear its `TransactionSignature` list here, making them eligible for GC, since they will never be used again.

---

But on this `stop-using-swirld-txn` branch, we:
 - Store the `TransactionSignature` objects [in the `PlatformTxnAccessor`](https://github.com/hashgraph/hedera-services/blob/stop-using-swirld-txn/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/utils/accessors/PlatformTxnAccessor.java#L57); which is,
 - [Stored](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/span/ExpandHandleSpan.java#L62) in the `Transaction#metadata` field; which is,
 - Already nulled out at consensus [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/span/ExpandHandleSpan.java#L71).

So on `stop-using-swirld-txn`, the `TransactionSignature`'s will **already** be eligible for GC after `handleConsensusRound()` completes, even if Platform keeps holding on the `ConsensusTransaction` for possible use in reconnect.

Reply:
That make sense ! Thanks for the explanation 😃